### ISSUE TITLE
memtest86+ is failing on staging (ld SIGSEGV)

### ISSUE THREAD

**Issue by globin** [NixOS/nixpkgs#27900]
## Issue description

cc @Ericson2314 

```
[nix-shell:~/dev/nixpkgs-staging/memtest86+-5.01]$ gdb /nix/store/0kf6dwhhlra1fxpvbfg6c2x6s3xi0ixz-binutils-2.29/bin/ld
GNU gdb (GDB) 8.0
Copyright (C) 2017 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-unknown-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /nix/store/0kf6dwhhlra1fxpvbfg6c2x6s3xi0ixz-binutils-2.29/bin/ld...(no debugging symbols found)...done.
(gdb) run -T memtest.bin.lds bootsect.o setup.o -b binary memtest_shared.bin -o memtest.bin
Starting program: /nix/store/0kf6dwhhlra1fxpvbfg6c2x6s3xi0ixz-binutils-2.29/bin/ld -T memtest.bin.lds bootsect.o setup.o -b binary memtest_shared.bin -o memtest.bin

Program received signal SIGSEGV, Segmentation fault.
0x00000000004273e4 in gldelf_x86_64_place_orphan ()
(gdb) bt
#0  0x00000000004273e4 in gldelf_x86_64_place_orphan ()
#1  0x000000000041829a in lang_process ()
#2  0x0000000000405c0e in main ()
```

**Comment by Ericson2314** [comment]
Wait this isn't memtest failing, but ld crashing when linking memtest?

**Comment by globin** [comment]
Yep /o\

**Comment by globin** [comment]
Fixed in 0e18f28cecd
